lib/zxing/client.rb

```diff
@@ -28,6 +23,21 @@ def self.responsive?(port)
       socket.close if socket
     end
 
+    def self.kill!
+      if remote_client
+        Process.kill(:INT, remote_client.pid)
+      end
+    end
+
+    private
+
+    def self.setup_drb_server(port)
+      @remote_client = IO.popen("#{ZXing::BIN} #{port}")
+
+      sleep 0.5 until responsive?(port)
```

Comment:
Should we have a max try to avoid freezing the process forever if something goes wrong?

Reply:
This code is from the original repository and never caused issue so I would rather not change it.